src/db_impl.cc

```diff
@@ -818,6 +835,24 @@ void TitanDBImpl::ReleaseSnapshot(const Snapshot* snapshot) {
   // We can record here whether the oldest snapshot is released.
   // If not, we can just skip the next round of purging obsolete files.
   db_->ReleaseSnapshot(snapshot);
+  {
+    MutexLock l(&mutex_);
+    if (pending_punch_hole_gc_ != nullptr && !punch_hole_gc_running_ &&
+        pending_punch_hole_gc_->snapshot()->GetSequenceNumber() ==
```

Comment:
I know we don't expect the pending sequence number to be smaller than `GetOldestSnapshotSequence()`, but do we consider this change, to be more defensive? 
```suggestion
        pending_punch_hole_gc_->snapshot()->GetSequenceNumber() <=
```